pkg/controller/sopssecret/sopssecret_controller.go

```diff
@@ -179,47 +142,40 @@ func (r *ReconcileSopsSecret) Reconcile(request reconcile.Request) (reconcile.Re
 		err = r.client.Get(
 			context.TODO(),
 			types.NamespacedName{
-				Name:      secret.Name,
-				Namespace: secret.Namespace,
+				Name:      newSecret.Name,
+				Namespace: newSecret.Namespace,
 			},
 			foundSecret,
 		)
-		if err != nil && errors.IsNotFound(err) {
-			reqLogger.Info(
-				"Creating a new Secret",
-				"Secret.Namespace",
-				secret.Namespace,
-				"Secret.Name",
-				secret.Name,
-			)
-			err = r.client.Create(context.TODO(), secret)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-
-			// Secret created successfully - don't requeue
-			return reconcile.Result{}, nil
-		} else if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Secret")
+			err = r.client.Create(context.TODO(), newSecret)
+			foundSecret = newSecret.DeepCopy()
+		}
+		if err != nil {
 			return reconcile.Result{}, err
-		} else {
-
-			// Secret already exists - enforce update
-			reqLogger.Info(
-				"Secret already exists: Update",
-				"Secret.Namespace",
-				foundSecret.Namespace,
-				"Secret.Name",
-				foundSecret.Name,
-			)
-			foundSecret.Labels = secret.Labels
-			foundSecret.Annotations = secret.Annotations
-			foundSecret.StringData = secret.StringData
+		}
+
+		if !metav1.IsControlledBy(foundSecret, instance) {
+			return reconcile.Result{}, fmt.Errorf("Secret isn't currently owned by sops-secrets-operator")
+		}
+
+		origSecret := foundSecret
+		foundSecret = foundSecret.DeepCopy()
 
+		foundSecret.StringData = newSecret.StringData
```

Comment:
This works, but it can cause some erroneous updates as it's converted to "Data" after it's already written. We may want to force everything into data and base64 encode so we can get a better comparison.